### ISSUE TITLE
Nx + React + Module Federation : ChunkLoadError: Loading chunk <name> failed

### ISSUE THREAD

**Issue by abhi9bakshi** [nrwl/nx#7527]
## TL;DR
Module federation works when container is Nx React app and Child is exposed from [CRA](https://github.com/facebook/create-react-app) with custom webpack config. Same child when exposed from Nx React app throws chunk load error because loader tries to load chunk from localhost:4200 (Container server port) instead of localhost:4201 (Child server port)

## Current Behavior
I have two apps in my Nx react project, viz. container and projects. The **projects app** exposes a button module using the following webpack configuration
```javascript
// apps/projects/webpack.config.js
const { ModuleFederationPlugin } = require('webpack').container;
const { dependencies } = require('../../package.json');

module.exports = (config, context) => {
  config.context = process.cwd();
  config.plugins.push(
    new ModuleFederationPlugin({
      name: 'app2',
      filename: 'remoteEntry.js',
      exposes: {
        './Button': './apps/projects/src/app/Button',
      },
      shared: {
        ...dependencies,
      },
    })
  );
  config.output.publicPath = 'http://localhost:4201/';

  return config;
};
```
The exposed module is made available in **container app** using the following config
```javascript
// apps/container/webpack.config.js
const { ModuleFederationPlugin } = require('webpack').container;
const nrwlConfig = require('@nrwl/react/plugins/webpack.js');

module.exports = (config, context) => {
  config.context = process.cwd();
  nrwlConfig(config);
  return {
    ...config,
    plugins: [
      ...config.plugins,
      new ModuleFederationPlugin({
        name: 'app1',
        remotes: {
          app2: `app2@${getRemoteEntryUrl(4201)}`,
        },
        shared: {
          react: { singleton: true },
          'react-dom': { singleton: true },
        },
      }),
    ],
  };
};

function getRemoteEntryUrl(port) {
  return `//localhost:${port}/remoteEntry.js`;
}
```

running the code results in error
```
Uncaught ChunkLoadError: Loading chunk default-apps_projects_src_app_Button_js failed.
(error: http://localhost:4200/default-apps_projects_src_app_Button_js.js)
while loading "./Button" from 3272
    at Object.__webpack_require__.f.j (jsonp chunk loading:27)
    at ensure chunk:6
    at Array.reduce (<anonymous>)
    at Function.__webpack_require__.e (ensure chunk:5)
    at Object../Button (container-entry:3)
    at get (container-entry:10)
    at handleFunction (remotes loading:33)
    at onInitialized (remotes loading:45)
    at handleFunction (remotes loading:38)
    at onExternal (remotes loading:44)
```

## Expected Behavior
If I change the webpack configuration of **container app** to
<pre>
// apps/container/webpack.config.js
const { ModuleFederationPlugin } = require('webpack').container;
const nrwlConfig = require('@nrwl/react/plugins/webpack.js');

module.exports = (config, context) => {
  config.context = process.cwd();

  nrwlConfig(config);
  return {
    ...config,
    plugins: [
      ...config.plugins,
      new ModuleFederationPlugin({
        name: 'app1',
        remotes: {
          <b>app2: `app2@${getRemoteEntryUrl(3002)}`,</b>
        },
        shared: {
          react: { singleton: true },
          'react-dom': { singleton: true },
        },
      }),
    ],
  };
};

function getRemoteEntryUrl(port) {
  return `//localhost:${port}/remoteEntry.js`;
}
</pre>
and clone [this repo which is forked from module federation examples and serve](https://github.com/abhi9bakshi/basic-host-remote), you will see the button module rendered in localhost:4200 url.

## Steps to Reproduce

1. Clone Nx repository
```
git clone https://github.com/abhi9bakshi/nx-react-microfrontend.git
```

2. Install dependencies
```
cd nx-react-microfrontend && npm install
```

3. Serve both applications
```
nx serve container --port 4200
nx serve projects --port 4201
```

4. Test container app in browser
```
http://localhost:4200
```
Here the container app will encounter error

5. Clone basic-host-remote app
```
git clone https://github.com/abhi9bakshi/basic-host-remote.git
```

6. Install dependencies
```
cd basic-host-remote/basic-host-remote/app2
npm install
```

7. Serve remote 
```
npm run start
```

8. Change `app2: app2@${getRemoteEntryUrl(4201)}` to `app2: app2@${getRemoteEntryUrl(3002)}` in apps/container/webpack.config.js

9. Test container app again in browser
```
http://localhost:4200
```
Here the container app will render button module successfully

### Environment
```yaml
  Node : 14.18.1
  OS   : darwin x64
  npm  : 6.14.15
  
  nx : Not Found
  @nrwl/angular : Not Found
  @nrwl/cli : 13.0.2
  @nrwl/cypress : 13.0.2
  @nrwl/devkit : 13.0.2
  @nrwl/eslint-plugin-nx : 13.0.2
  @nrwl/express : Not Found
  @nrwl/jest : 13.0.2
  @nrwl/linter : 13.0.2
  @nrwl/nest : Not Found
  @nrwl/next : Not Found
  @nrwl/node : Not Found
  @nrwl/nx-cloud : Not Found
  @nrwl/react : 13.0.2
  @nrwl/schematics : Not Found
  @nrwl/tao : 13.0.2
  @nrwl/web : 13.0.2
  @nrwl/workspace : 13.0.2
  @nrwl/storybook : 13.0.2
  @nrwl/gatsby : Not Found
  typescript : 4.3.5
```

**Comment by abhi9bakshi** [comment]
This issue is solved by adding optimization and output params to webpack configuration. The latest code is available at branch [fix](https://github.com/abhi9bakshi/nx-react-microfrontend/tree/fix)

```diff
const { ModuleFederationPlugin } = require('webpack').container;
const { dependencies } = require('../../package.json');
const path = require("path");

module.exports = (config, context) => {
  config.context = process.cwd();
+  config.optimization.runtimeChunk = false;
  config.plugins.push(
    new ModuleFederationPlugin({
      name: 'app2',
      filename: 'remoteEntry.js',
      exposes: {
        './Button': './apps/projects/src/app/Button',
      },
      shared: {
        react: { singleton: true },
        'react-dom': { singleton: true },
      },
    })
  );
+  config.output = {
+    uniqueName: 'app2',
+   publicPath: 'auto',
+    clean: true,
+  }
  return config;
};
```
